// sg_employee_module/MyPackageAnonymousModule.js
//All anonymous modules are exported by external applications through modules.exports
//module.exports is a nodejs object that exports the components for usability.
//Usually anonymous modules are self invoking modules.

//second parnthesis is for method invocation.


//var emp = require('./employee');
module.exports = (function(){
	var repository = [];

	function addNewEmployee(emp){
		repository.push(emp);
	}

	function deleteEmployee(id){
	 var emp = repository.find(function(element){
			return element.empId == id;
		})
	 var index = repository.indexId(emp);
	 repository.splice(index,1);
	}

	function getAllEmployees(){
		return repository;
	}

	function updateEmployee(emp){
		for(i=0;i<repository.length;i++){
			if(repository[i].empId == emp.empId){
				repository[i].empName = emp.empName;
				return;
			}
		}
		throw "Employee not found to update";
	}

	return {
		addNew : addNewEmployee,
		deleteEmployee : deleteEmployee,
		update : updateEmployee,
		getAll : getAllEmployees
	}

})();
